include/exec/async_scope.hpp

```diff
@@ -31,6 +31,14 @@ namespace exec {
     struct __impl;
     struct async_scope;
 
+    template<class _A> 
+    concept __async_scope = 
+      !std::is_copy_constructible_v<_A> && !std::is_move_constructible_v<_A> && 
+      !std::is_copy_assignable_v<_A> && !std::is_move_assignable_v<_A> && 
```

Comment:
if a type has a `.nest()` member that returns a `sender_of<set_value_t()>`, do you really want the algorithms to reject the type just because it's movable? how is immovability an essential property of async scopes?